File: 09243.py
N = int(input())
before = input()
after = input()

wrong = False
if N % 2:
    for i in range(len(before)):
        if before[i] == after[i]:
            print("Deletion failed")
            wrong = True
            break
else:
    for i in range(len(before)):
        if before[i] != after[i]:
            print("Deletion failed")
            wrong = True
            break

if not wrong: print("Deletion succeeded")

File: 09243.cpp
#include <iostream>
using namespace std;

int main() {
	int N; cin >> N;
	string before, after;
	cin >> before >> after;

	bool wrong = false;
	// 홀수면 반대로
	if (N % 2) {
		for (int i = 0; i < before.length(); i++) {
			if (before[i] == after[i]) {
				cout << "Deletion failed\n";
				wrong = true;
				break;
			}
		}
	}
	// 짝수면 그대로
	else {
		for (int i = 0; i < before.length(); i++) {
			if (before[i] != after[i]) {
				cout << "Deletion failed\n";
				wrong = true;
				break;
			}
		}
	}

	if (!wrong) cout << "Deletion succeeded\n";
}
